src/tests/src/solver/modelParser/testConvertorVisitor.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/solver/expressions/Registry.hxx"
+#include "antares/solver/expressions/visitors/AstDOTStyleVisitor.h"
+#include "antares/solver/expressions/visitors/CompareVisitor.h"
+#include "antares/solver/libObjectModel/library.h"
+#include "antares/solver/modelConverter/convertorVisitor.h"
+#include "antares/solver/modelConverter/modelConverter.h"
+#include "antares/solver/modelParser/Library.h"
+
+using namespace Antares::Solver;
+
+struct Fixture
+{
+    ModelParser::Model model;
+    Antares::Solver::Registry<Antares::Solver::Nodes::Node> registry;
+};
+
+static Nodes::LiteralNode* toLiteral(Nodes::Node* n)
+{
+    return dynamic_cast<Nodes::LiteralNode*>(n);
+}
+
+BOOST_FIXTURE_TEST_CASE(empty_expression, Fixture)
+{
+    auto* node = ModelConverter::convertExpressionToNode("", registry, model);
```

Comment:
About the **registry** : 
As you know, the **registry** comes with a tree, it makes sure we avoid memory leaks when destroying the tree (so, a registry and the associated tree should have the same life time and be more dependent, but this another matter).
On the other hand, we have unit tests converting an expression into a tree and checking the created tree.
These unit tests (and any calling code) don't have to and don't want to see the registry.
**Suggestion** : 
instead of a function, make the converter a **class**, and define an instance of this class in the **fixture**, and run the converter in the tests : 
```C++
struct Fixture
{
    ModelParser::Model model;
    ModelConverter::ExpressionToNodeConverter converter(model); // Registry created internally
};
...

BOOST_FIXTURE_TEST_CASE(empty_expression, Fixture)
{
    auto* node = converter.run("");
    BOOST_CHECK_EQUAL(node, nullptr);
}
...
```